hyperspy/io_plugins/tiff.py

```diff
@@ -451,9 +438,188 @@ def wave_len(ht):
         scale /= camera_len * wave_len(ht) * 1e9  # in nm
         scales['x'], scales['y'] = _get_scales_from_x_y_resolution(op, factor = scale)
         units = {"x": "1 / nm", "y": "1 / nm", "z": t.Undefined}
-    return scales, units, offsets, intensity_axis
+    return scales, offsets, units, intensity_axis
 
-def _get_scales_from_x_y_resolution(op, factor=1):
+
+def _is_streak_hamamatsu(op) -> bool:
+    """Determines whether a .tiff page is likely to be a hamamatsu
+    streak file based on the original op content.
+    """
+    is_hamatif = True
+
+    # Check that the original op has an "Artist" that copyrights as hamamatsu
+    if not 'Artist' in op:
+        is_hamatif = False
+        return is_hamatif
+    else:
+        artist = op['Artist']
+        if not artist.startswith("Copyright Hamamatsu"):
+            is_hamatif = False
+            return is_hamatif
+
+    # Check that the original op has a "Software" corresponding to the HPD-TA
+    if not 'Software' in op:
+        is_hamatif = False
+        return is_hamatif
+    else:
+        software = op['Software']
+        if not software.startswith('HPD-TA'):
+            is_hamatif = False
+
+    return is_hamatif
+def _get_hamamatsu_streak_description(tiff, op):
+    """Extract a dictionary recursively from the ImageDescription
+    Metadata field in a Hamamatsu Streak .tiff file"""
+    import csv
+    desc = op['ImageDescription']
+    dict_meta = {}
+    reader = csv.reader(desc.splitlines(), delimiter=',', quotechar='"')
+    for row in reader:
+        key = row[0].strip(" []")
+        key_dict = {}
+        for element in row[1:]:
+            spl = element.split('=')
+            if len(spl) == 2:
+                key_dict[spl[0]] = spl[1].strip('"')
+        dict_meta[key] = key_dict
+
+    import re
```

Comment:
Imports in the middle of functions are quite unusual. Should at least be at the top of the method, but I think they are generally at the top of the files across HyperSpy,

Reply:
Ok, the idea was to run import only if the .tiff is recognized as hamamatsu but fair enough it doesn't matter much